src/backends/github/API.js

```diff
@@ -229,50 +208,26 @@ export default class API {
   }
 
   composeFileTree(files) {
-    let filename;
-    let part;
-    let parts;
-    let subtree;
-    const fileTree = {};
-
-    files.forEach((file) => {
-      if (file.uploaded) { return; }
-      parts = file.path.split("/").filter(part => part);
-      filename = parts.pop();
-      subtree = fileTree;
-      while (part = parts.shift()) {
-        subtree[part] = subtree[part] || {};
-        subtree = subtree[part];
-      }
-      subtree[filename] = file;
-      file.file = true;
-    });
-
-    return fileTree;
+    return files
+      .map(file => ({ ...file, file: true }))
+      .reduce((tree, file) => tree.setIn(file.path.split("/"), file), Map())
+      .toJS();
```

Comment:
😍

Reply:
Does it matter that the original version did not include nested keys that were empty and the new one does?
I.e. `.filter(part => part)` appears to have been dropped.